From 4d73aef9d781d327d5b68b07c2b477f66244730a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Thu, 25 Oct 2018 13:15:10 +0800
Subject: [PATCH 4933/5242] MLK-20090: ARM64: dts: enable headphone detection
 for imx8qm mek

commit  6464af78aa194d4c81054394732afea260f68b45 from
https://source.codeaurora.org/external/imx/linux-imx.git

In SPF-29420-B2, the SCU_GPIO0_03 is used as AUD_DEC_1V8 for wm8960
codec.  With this chage, the headphone plugin/plugout detection
can be enabled.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index f475c42..8181f2f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -133,6 +133,8 @@
 		 * hp-det-polarity = 1: hp detect high for speaker
 		 */
 		hp-det = <2 0>;
+		hp-det-gpios = <&gpio0 31 0>;
+		mic-det-gpios = <&gpio0 31 0>;
 		audio-routing =
 			"Headphone Jack", "HP_L",
 			"Headphone Jack", "HP_R",
@@ -249,6 +251,7 @@
 			fsl,pins = <
 				SC_P_MCLK_OUT0_AUD_ACM_MCLK_OUT0	0x0600004c
 				SC_P_QSPI1A_DATA1_LSIO_GPIO4_IO25	0x0600004c
+				SC_P_SCU_GPIO0_03_LSIO_GPIO0_IO31	0x0600004c
 			>;
 		};
 
-- 
1.7.9.5

